posydon/unit_tests/utils/test_posydonerror.py

```diff
@@ -22,113 +39,144 @@ def test_dir(self):
+            raise totest.POSYDONError("Test")
 
     def test_instance_FlowError(self):
-        self.assertTrue(isclass(totest.FlowError))
-        self.assertTrue(issubclass(totest.FlowError, totest.POSYDONError))
+        assert isclass(totest.FlowError)
+        assert issubclass(totest.FlowError, totest.POSYDONError)
+        with raises(totest.FlowError, match="Test"):
+            raise totest.FlowError("Test")
 
     def test_instance_GridError(self):
-        self.assertTrue(isclass(totest.GridError))
-        self.assertTrue(issubclass(totest.GridError, totest.POSYDONError))
+        assert isclass(totest.GridError)
+        assert issubclass(totest.GridError, totest.POSYDONError)
+        with raises(totest.GridError, match="Test"):
+            raise totest.GridError("Test")
 
     def test_instance_MatchingError(self):
-        self.assertTrue(isclass(totest.MatchingError))
-        self.assertTrue(issubclass(totest.MatchingError, totest.POSYDONError))
+        assert isclass(totest.MatchingError)
+        assert issubclass(totest.MatchingError, totest.POSYDONError)
+        with raises(totest.MatchingError, match="Test"):
+            raise totest.MatchingError("Test")
 
     def test_instance_ModelError(self):
-        self.assertTrue(isclass(totest.ModelError))
-        self.assertTrue(issubclass(totest.ModelError, totest.POSYDONError))
+        assert isclass(totest.ModelError)
+        assert issubclass(totest.ModelError, totest.POSYDONError)
+        with raises(totest.ModelError, match="Test"):
+            raise totest.ModelError("Test")
 
     def test_instance_NumericalError(self):
-        self.assertTrue(isclass(totest.NumericalError))
-        self.assertTrue(issubclass(totest.NumericalError, totest.POSYDONError))
+        assert isclass(totest.NumericalError)
+        assert issubclass(totest.NumericalError, totest.POSYDONError)
+        with raises(totest.NumericalError, match="Test"):
+            raise totest.NumericalError("Test")
 
     def test_instance_initial_condition_message(self):
-        self.assertTrue(isroutine(totest.initial_condition_message))
+        assert isroutine(totest.initial_condition_message)
 
 
-class TestFunctions(unittest.TestCase):
+class TestFunctions:
     # test functions
-    def setUp(self):
-        # initialize a BinaryStar instance, which is a required argument
-        self.BinaryStar = totest.BinaryStar()
-
-    def test_initial_condition_message(self):
-        test_object = {'Test': 'object'}
-        with self.assertRaises(TypeError):
-            message = totest.initial_condition_message(binary=test_object)
-        self.assertIn("Failed Binary Initial Conditions",
-            totest.initial_condition_message(binary=self.BinaryStar))
-        with self.assertRaises(TypeError):
-            message = totest.initial_condition_message(binary=test_object,
+    def test_initial_condition_message(self, BinaryStar, artificial_object):
```

Comment:
Sure, those are multiple tests on the same function. It will be a lot more work to split up any test into a single function. I put the line to one test per function. Splitting this up into tests for errors and tests for outputs, will cause people to shift things around on subjective grounds, e.g. an error captured in a try causing an output -- would that belong to outputs or to errors?. Hence, the other objective way I'd have in mind would be to make limit test functions to have not more then one assert. This will be a lot more work and less efficient because you can't reuse stuff.
So, if you have a better objective way to split up tests, please feel free to propose.

Reply:
I agree that one assert per test is overkill and not worth it.
For example, testing if variables are correctly set on an __init__ should be fine.

I'm a bit split about how detailed these should be though. I find these a bit too unique to be in one test. But that's subjective too.

Reply:
I tried to use some comments to give some block structure inside the test functions.